.github/workflows/update_version.yml

```diff
@@ -20,54 +20,36 @@ jobs:
       - name: Get current date for the first release candidate
         if: github.event_name == 'schedule'
         run: |
-          echo "VERSION=$(date +'v%y.%-m')rc0" >> $GITHUB_ENV
-          echo "NON_RC_VERSION=$(date +'v%y.%-m')" >> $GITHUB_ENV
+          echo "VERSION=$(date +'v%y.%-m' + ".0")" >> $GITHUB_ENV
 
       - name: Get current date for a manual release
         if: github.event_name == 'workflow_dispatch'
         run: |
           echo "VERSION=$(date +'v%y.%-m')${{ github.event.inputs.append_to_tag }}" >> $GITHUB_ENV
-          echo "NON_RC_VERSION=$(date +'v%y.%-m')" >> $GITHUB_ENV
 
       # the schedule workflow is for rc0 release
       - uses: actions/checkout@v4
-        if: github.event_name == 'schedule'
         with:
           ref: 'develop'
 
-      # the dispatch workflow is for rcX and final releases
-      - uses: actions/checkout@v4
-        if: github.event_name == 'workflow_dispatch'
-        with:
-          ref: '${{ env.NON_RC_VERSION }}'
```

Comment:
Why are we not checking out to the version branch for bug fixes? Checking out to develop will bring on all the changes, including any new features that went in.

Reply:
Replaced